core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -73,11 +73,11 @@ export class StateGraphLayoutService {
 
   // The last result of a call to computeLayout(). Used for determining the
   // order in which to specify states in rules.
-  lastComputedArrangement: NodeDataDict = null;
+  lastComputedArrangement: NodeDataDict | undefined;
```

Comment:
Why this change? Why not
```suggestion
  lastComputedArrangement: NodeDataDict | null = null;
```

Reply:
well I am not sure about this.... we can do that but then we need to add 
`getLastComputedArrangement(): NodeDataDict | null {`   - at line 488 
instead of  `getLastComputedArrangement(): NodeDataDict | undefined {` which I committed.

Can you please tell me the difference between undefined and null over here? I am not able to get this

Reply:
They are basically the same, you can read some resources online. I think in our codebase we usually prefer to use `null` in cases where we return something that is logically empty. `undefined` usually means that something is missing (i.e. not defined).